pkg/linter/linter.go

```diff
@@ -152,10 +152,38 @@ func New(cxt *portercontext.Context, mixins pkgmgmt.PackageManager) *Linter {
 	}
 }
 
+
 func (l *Linter) Lint(ctx context.Context, m *manifest.Manifest) (Results, error) {
 	// TODO: perform any porter level linting
 	// e.g. metadata, credentials, properties, outputs, dependencies, etc
+	reservedPrefixes := []string{"porter-", "porter_"}
+	params := m.Parameters
+
+	var results Results
 
+	for _, param := range params {
+		paramName := strings.ToLower(param.Name)
+		for _, reservedPrefix := range reservedPrefixes {
+			if strings.HasPrefix(paramName, reservedPrefix) {
+
+				res := Result {
+					Level: LevelWarning,
+					Location: Location {
+						Action: "",
+						Mixin: "",
+						StepNumber: 0,
+						StepDescription: "",
+				
+					},
+					Code: "exec-100",
```

Comment:
Let's give this one a unique code, `porter-100` so that it doesn't get confused with lint results returned from the exec mixin.